examples/README.md

```diff
@@ -0,0 +1,132 @@
+# Overview
+We are providing an open-source illustration and demonstration of example/demo. This will help us to understand the usage of the concord-bft library. Since concord-bft is an open-source consensus, this application will also assist us in comprehending the internals of concord-bft.
+
+It is responsible for assisting and guiding large number of concord-bft users around the globe, including those who want to understand more about concord-bft, its usage, and how it works.
+
+This example/demo will provide end to end functionality of Concord byzantine fault tolerant state machine replication library about its usage. This can be used as a library to build a distributed trust infrastructure that is extremely scalable and low-power for smart contract execution.
+
+**Note:** Initially, we plan to do this using an open source key-value blockchain replica. However, once the setup is finished, we'll also use alternative open source execution engines, such Ethereum, WASM, etc.
+
+
+## Install and Build (Ubuntu Linux 18.04 or above)
+Concord-BFT supports two kinds of builds are,
+* Docker
+* Native
+
+The docker build is **strongly recommended**
+
+
+### Docker
+
+* [Install the latest docker] (https://docs.docker.com/engine/install/ubuntu/).
+* Optional: [configure docker as non-root user](https://docs.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user).
+* Build concord-bft
+```sh
+git clone https://github.com/vmware/concord-bft
+cd concord-bft
+make
+```
+Run `make help` to see more commands.
+
+Note:
+* If you face any format related issues than run `make format` before running `make` command to format the changes.
+* The output binaries are stored in the host's `concord-bft/build`.
+* Default compiler is clang and clang++.
+* `Makefile` is configurable. If you want to use another compiler you may pass it to the `make`.
+For example,
+```
+make CONCORD_BFT_CONTAINER_CXX=g++ \
+    CONCORD_BFT_CONTAINER_CC=gcc \
+    build
+```
+Other build options, including passthrough options for CMake, are defined in the [Makefile](../Makefile) and prefixed with `CONCORD_BFT_`.
+
+
+### Native
+
+For native development environment, need to install some dependencies mentioned in [install_deps.sh](../install_deps.sh) script.
+```sh
+git clone https://github.com/vmware/concord-bft
+cd concord-bft
+sudo ./install_deps.sh    # Install all dependencies and 3rd parties
+mkdir build
+cd build
+cmake ..
+make
+```
+
+
+## Run Example/Demo
+This simple example can be run from the script [test_example.sh](scripts/test_example.sh) under `example/script` directory.
+
+
+### Simple scenario (4 replicas and 1 client)
+This demo can be run from inside the container once the build is finished.
+
+Run the [test_example.sh](scripts/test_example.sh) script from the container's `build/example/scripts` directory.
```

Comment:
To make sure the demo isn't regressed by ongoing code changes, it may be a good idea to add a (nightly?) CI workflow that makes sure it works as expected.

Reply:
PS: this can be done in a separate PR, of course.

Reply:
Yeah it is in progress.